### PR TITLE
[Backend] Allow capital letters and underscore in metric names

### PR DESCRIPTION
**Pull request by Bobgy**
Fixes https://github.com/kubeflow/pipelines/issues/3655
Mitigates https://github.com/kubeflow/pipelines/issues/3499
Doc PR in https://github.com/kubeflow/website/pull/1950

FYI, why I did this: https://github.com/kubeflow/pipelines/issues/3499#issuecomment-627132988

Verified it works e2e using test image gcr.io/ml-pipeline-test/d73087f0a2d71aefc891cb269545c3062b9abf0a/api-server@sha256:3b4a1503412e930962e886856abf2fbea6ac31772abcc6aa54072dfef20fe66d
Screen Shot 2020-05-12 at 4 25 15 PM

/assign @jingzhang36 
/cc @amygdala @numerology

### CODE REVIEW

File: backend/src/apiserver/server/run_metric_util.go [hunk 1]
```diff
@@ -24,7 +24,12 @@ import (
 )
 
 const (
-	metricNamePattern = "^[a-z]([-a-z0-9]{0,62}[a-z0-9])?$"
+	// This regex expresses the following constraints:
+	// * Allows lowercase/capital letters
```
Comment (jingzhang36) [inline backend/src/apiserver/server/run_metric_util.go:28]: lowercase/uppercase just to be symmetric?

Comment (Bobgy) [reply backend/src/apiserver/server/run_metric_util.go:28]: good point, I forgot about uppercase. lol

**Pull request merged**